resolver-dns/src/main/java/io/netty5/resolver/dns/LoggingDnsQueryLifecycleObserver.java

```diff
@@ -46,38 +46,38 @@ public void queryWritten(InetSocketAddress dnsServerAddress, Future<Void> future
     @Override
     public void queryCancelled(int queriesRemaining) {
         if (dnsServerAddress != null) {
-            logger.log(level, "from {} : {} cancelled with {} queries remaining", dnsServerAddress, question,
+            logger.atLevel(level).log("from {} : {} cancelled with {} queries remaining", dnsServerAddress, question,
```

Comment:
Could extract `logger.atLevel(level)` into a member field so we don't have to call it every time. I guess this could be potentially creating a new instance for each log message.

Reply:
If we cache the builder we will also not be told about runtime log level changes.